Add login page UI

nmartindev opened the pull request:
This pull request contains the following:

- Create a basic TextTheme and colorScheme
- Create UI for logging in

The current UI looks like this:

![login-page](https://github.com/user-attachments/assets/e327759b-b7d1-4213-a584-697c10164740)

nmartindev committed:
Create login page

nmartindev committed:
Add a SafeArea

nmartindev committed:
Create textTheme

nmartindev committed:
Create color theme and OutlinedButton theme

nmartindev committed:
Add SSO login buttons

nmartindev committed:
Add elevated button theme

nmartindev committed:
Add sign in button

nmartindev committed:
Add TextButtons and theme

nmartindev committed:
Fix UI overflow

MathNerd28 commented on lib/assets/fonts/Montserrat-Bold.ttf:
Flutter assets typically don't go in the `lib` directory, that's just for dart code

MathNerd28 commented on lib/pages/login.dart:
Overall, the UI looks pretty good here. Couple of notes:
- Google's guidelines state ["The button must always include the standard color for the Google 'G'"](https://developers.google.com/identity/branding-guidelines#color). Black and white is not okay.
- The only non-blue color on the screen is the hand waving at the top, which is constantly drawing my attention when it really shouldn't.
- The sign-in button should not be actionable when either the email or password fields are empty... but we'll get to that when adding controllers and logic.

The page is also a bit cramped... one possible idea is to break the email/password fields out to a separate page, and have a "sign in with email/password" button in their place. One extra benefit of this is that same page can be reused for creating accounts with email/password, and it further encourages using SSO during onboarding.

MathNerd28 commented on lib/pages/login.dart:
```diff
@@ -0,0 +1,154 @@
+                          padding: const EdgeInsets.only(bottom: 6.0),
+                          child: Text(
+                            "Welcome! 👋",
+                            style: Theme.of(context).textTheme.displayMedium,
```
14 levels of indentation is ridiculous, even for Flutter. To make this more readable, you could break out methods for each of the major UI elements, like so:
```dart
children: [
  _childOne(context),
  _childTwo(context),
]
```
```dart
Widget _childOne(BuildContext context) => YourWidgetHere();
```
This will make it easier to find and edit specific elements, or reorder them on the page.

MathNerd28 replied:
For example, this page might look something like:
```dart
children: [
  _welcomeText(context),
  _signInWithGoogle(context),
  _signInWithApple(context),
  _orDivider(context),
  _emailField(context), // combining header and text field
  _passwordField(context), // combining header, text field, and 'forgot password'
  _signInButton(context),
  _createAccountText(context),
]
```

MathNerd28 reviewed (changes requested):
Overall, really nice work. A few things to fix, and some suggestions. I love this UI style, and can't wait to see more of it.

nmartindev committed:
Move assets folder

nmartindev committed:
Clean code and UI

nmartindev committed:
Change wording of text

nmartindev committed:
Reorder code

nmartindev committed:
Remove ScrollView

nmartindev committed:
Change font to Noto Sans

nmartindev committed:
Change wording again

nmartindev committed:
Rename login.dart to login_select.dart

nmartindev committed:
Add sign in info button

nmartindev committed:
Create email sign in page

nmartindev committed:
Add sign up page

nmartindev committed:
Add verify password input

nmartindev committed:
Update ElevatedButton overlay

nmartindev commented:
**After many changes, here is the current login UI:**

![login-select](https://github.com/user-attachments/assets/80c77560-a9fc-4eb1-a518-c094bf50e8fa) ![email-login](https://github.com/user-attachments/assets/72595b98-917d-43cc-8095-d56881f493bf) ![signup-page](https://github.com/user-attachments/assets/9c3f207d-6b8b-430e-ad92-a13b43c9075f)

**Changes Made:**

- Cleaned code and file structure, making it more readable and organized
- Corrected the use of icons and brand logos
- Revised wording for more clarity
- Changed font for better UI appeal and clarity
- Restructured the Sign In Method Select page to remove clutter
- Move email sign in to a new page
- Create a page for new user sign up (with email; SSO is automatic if an account does not exist)
- Update theming to be more cohesive

nmartindev committed:
Add name field

nmartindev committed:
Create custom component for data fields

nmartindev committed:
Re-add forgot password button

nmartindev committed:
Change navigation path between pages

MathNerd28 reviewed (dismissed):
Looks great! Love the code cleanup

MathNerd28 committed:
Merge branch 'main' into login-page

Signed-off-by: Alexander Bhalla <xander.bhalla@gmail.com>
